lms/static/js/commerce/track_ecommerce_events.js

```diff
@@ -0,0 +1,40 @@
+/**
+ *
+ * A library of helper functions to track ecommerce related events. 
+ *
+ */
+
+(function(define) {
+    'use strict';
+
+    define([
+        'jquery'
+    ], function($) {
+        var trackUpsellClick = function(linkName, optionalAttrs) {
+            if (!window.analytics) {
+                return;
+            }
+    
+            var eventAttrs = { "linkName": linkName };
+
+            if (
+                typeof optionalAttrs !== 'undefined' &&
+                optionalAttrs !== null &&
+                Object.keys(optionalAttrs).length > 0
+            ) {
+                var allowedAttrs = ["courseId", "pageName"];
+                allowedAttrs.map(
+                    allowedAttr => eventAttrs[allowedAttr] = optionalAttrs[allowedAttr]
```

Comment:
Can't actually use arrow functions here.  This code won't be transpiled.
```
allowedAttrs.map(function(allowedAttr) {
    eventAttrs[allowedAttr] = optionalAttrs[allowedAttr];
}));
```

Reply:
Updated